src/app/core/pages/Authcomponents/login/login.component.ts

```diff
@@ -0,0 +1,52 @@
+import { validsignup } from './../../../../shared/utilites/validsignup';
+import { Subject, takeUntil } from 'rxjs';
+import { FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
+import { Component, inject } from '@angular/core';
+import { HttpErrorResponse } from '@angular/common/http';
+import { Router } from '@angular/router';
+import { IconComponent } from '../../../../shared/components/ui/icon/icon.component';
+import { EmailInputComponent } from '../../../../shared/components/ui/email-input/email-input.component';
+import { AlertsComponent } from '../../../../shared/components/ui/alerts/alerts.component';
+import { RegButtonComponent } from '../../../../shared/components/ui/reg-button/reg-button.component';
+import { BackgroundComponent } from '../../../../shared/components/ui/background/background.component';
+import { AuthApiService } from '../../../../../../projects/auth-api/src/public-api';
+
+
+@Component({
+  selector: 'app-login',
+  imports: [IconComponent, EmailInputComponent, ReactiveFormsModule,AlertsComponent,RegButtonComponent , BackgroundComponent],
+  templateUrl: './login.component.html',
+  styleUrl: './login.component.scss'
+})
+export class LoginComponent {
+    private ngUnsubscribe = new Subject<void>(); 
+  
+  private _router=inject(Router)
```

Comment:
```suggestion
  private _router=inject(Router)
```
don't using private use the consterctor